kidviz/views.py

```diff
@@ -69,7 +77,17 @@ def form_valid(self, form):
         self.request.session['constructs'] = [c.id for c in constructs]
         self.request.session['context_tags'] = [t.id for t in tags]
         self.request.session['curricular_focus'] = d['curricular_focus']
-        self.request.session['create_new'] = True
+
+        if 're_setup' in self.request.session:
```

Comment:
`if self.request.session.pop('re_setup')` ot sth.

Reply:
I have to check for the key otherwise `Key error` may be raised if key doesn't exist. Should i use try-except block or rather stick with if?

Reply:
Check if you can pass default response e.g.  pop('re_serup', None). Otherwise its OK.

Reply:
I mean: stay with `if` and pop value in first if branch,